src/main/java/ch/carve/microprofile/config/db/DatasourceConfigSource.java

```diff
@@ -38,11 +46,17 @@ public String getValue(String propertyName) {
         TimedEntry entry = cache.get(propertyName);
         if (entry == null || entry.isExpired()) {
             log.debug("load {} from db", propertyName);
-            String value = repository.getConfigValue(propertyName);
-            cache.put(propertyName, new TimedEntry(value));
-            return value;
+            try {
+                String value = repository.getConfigValue(propertyName);
+                cache.put(propertyName, new TimedEntry(value));
+                return value;
+            } catch (SQLException e) {
+                log.info("query failed: " + e.getMessage());
+                clearRepository();
+            }
         }
-        return entry.getValue();
+        // if the entry was never cached, then it will be null
+        return entry != null ? entry.getValue() : null;

```

Comment:
then leave this as is

Reply:
yeaph, thanks for pointing to a cleaner solution

Reply:
as just found out: If the entry is expired and the sql fails, then reuse the expired value.